### PR TITLE
First pass at a full terraform example

### PR DESCRIPTION
**Pull request by regularfry**
## Description

As a general rule, the repository template should provide functionality that works out of the box.  The terraform example is problematic because the options default to running the example, and the example doesn't give any guidance as to how people should set up their own code.

This patch removes the example, and replaces it with a more fully explained quick start example in `Scripting_Terraform.md`.

It also adds the `TF_ENV` environment variable, as a shorthand for selecting the terraform environment directory under `infrastructure/environments`.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: scripts/terraform/terraform.mk [hunk 1]
```diff
@@ -55,23 +58,6 @@ terraform-shellscript-lint: # Lint all Terraform module shell scripts @Quality
 		file=$${file} scripts/shellscript-linter.sh
 	done
 
-# ==============================================================================
```
Comment (stefaniuk) [inline scripts/terraform/terraform.mk:58]: Can we also remove along with the example the following, please:

```
	terraform-example-clean \
	terraform-example-destroy-aws-infrastructure \
	terraform-example-provision-aws-infrastructure \
```

This is in the `${VERBOSE}.SILENT` section at the bottom of the file.

**Review by stefaniuk — CHANGES_REQUESTED** [review]
This PR looks good to me. A guide reads better than a dry example.

Just one small additional change would be nice to make to clean up.

File: scripts/terraform/terraform.mk (hunk 1 above)
Comment (regularfry) [reply scripts/terraform/terraform.mk:58]: Good catch - fixed.

**Review by stefaniuk — APPROVED** [review]
LGTM

**Pull request merged**